datastore/README.md

```diff
@@ -0,0 +1,76 @@
+General-purpose key-value storage.
+The keys are strings, and the values are of any type you want.
+
+> **Note**: This crate is meant to be a utility for the implementation of other crates ; it
+>			does not directly participate in the stack of libp2p.
+
+This crate provides the `Datastore` trait, whose template parameter is the type of the value.
+It is implemented on types that represent a key-value storage.
+The only available implementation for now is `JsonFileDatastore`.
+
+# JSON file datastore
+
+The `JsonFileDatastore` can provide a key-value storage that loads and stores data in a single
+JSON file. It is only available if the value implements the `Serialize`, `DeserializeOwned`
+and `Clone` traits.
+
+The `JsonFileDatastore::new` method will attempt to load existing data from the path you pass
+as parameter. This path is also where the data will be stored. The content of the store is
+flushed on destruction or if you call `flush()`.
```

Comment:
Probably use "on drop" instead of "on destruction" to make it clearer what you mean.